lib/generators.nix

```diff
@@ -177,21 +177,24 @@ rec {
 
   # PLIST handling
   toPlist = {}: v: let
-    expr = ind: x: with builtins;
-      if isNull x then "" else
-      if isBool x then bool ind x else
-      if isInt x then int ind x else
+    isFloat = builtins.isFloat or (x: false);
+    expr = ind: x:  with builtins;
+      if isNull x   then "" else
+      if isBool x   then bool ind x else
+      if isInt x    then int ind x else
       if isString x then str ind x else
-      if isList x then list ind x else
-      if isAttrs x then attrs ind x else
+      if isList x   then list ind x else
+      if isAttrs x  then attrs ind x else
+      if isFloat x  then float ind x else
       abort "generators.toPlist: should never happen (v = ${v})";
 
     literal = ind: x: ind + x;
 
     bool = ind: x: literal ind  (if x then "<true/>" else "<false/>");
-    int = ind: x: literal ind "<integer>${toString x}</integer>";
```

Comment:
I'm pretty sure this is necessary, same with floats.

Reply:
Really? I'll see what happens but I thought that ${} would convert it to a string automatically.